### DISCUSSION TITLE
MediaFoundation: Get supported formats from IMFMediaType

### DISCUSSION THREAD

**Discussion by anguzo** [amerkoleci/Vortice.Windows#382] [Ideas]
Hi!

I want to contribute to project that is using Vortice - Flyleaf, by adding enumeration of capture device supported formats. Since it is already using Vortice.MediaFoundation for device enumeration(MFEnumDeviceSources), I thought that adding this feature using Media Foundation would be most logical. I started getting into Media Foundation documentation and found `https://github.com/OfItselfSo/Tanta` samples very useful but I found that some methods needed for that are missing from Vortice.MediaFoundation.
These methods are:
1. MFGetAttributeSize
2. MFGetAttributeRatio

There are also other GetAttribute methods.

I  actually managed to replace both these methods and get needed format information using bit manipulation like so:
```
var frameRateUlong = mediaTypeObj.Get<ulong>(MediaTypeAttributeKeys.FrameRate);

frameRate = (int)(frameRateUlong >> 32);
frameRateDenominator = (int)(frameRateUlong & 0xFFFFFFFF);
```
But it would be nice if I could do it similar to how Media Foundation documentation suggests:
```
MediaFactory.MFGetAttributeRatio(mediaTypeObj, MediaTypeAttributeKeys.FrameRate, out frameRate, out frameRateDenominator)
```

Thank you!

**Comment by amerkoleci** [comment]
Thanks for tips, just added those helpers:
https://github.com/amerkoleci/Vortice.Windows/commit/b62109059c6704dcb106bd32060ba50c49bd5f03
